includes/sync.php

```diff
@@ -65,14 +67,17 @@ function sync_shadow_taxonomies( int $post_id, \WP_Post $post_after, bool $updat
 		return;
 	}
 
-	// If a post is to remain published, but the title has changed, update the term.
-	if ( $term_before && 'publish' === $post_after->post_status && $title_before !== $title_after ) {
+	$changed = $title_before !== $title_after;
+	$changed = $changed || ( $slug_before !== $slug_after );
```

Comment:
```suggestion
	$changed = $title_before !== $title_after || $slug_before !== $slug_after;
```
This is probably split for a reason, but suggesting just in case this one-line version works 🙂